### PR TITLE
Move to Pixi and Hatchling

### PR DESCRIPTION
**Pull request by hoxbro**
Difference in build:

``` yaml
                                Lumen - wheel
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Files only in run 1                 ┃ Files only in run 2                 ┃
┃ lumen 0.7.0a11 py2.py3 none any.whl ┃ lumen 0.7.0a11.post1.dev46+gce9c313 ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ $VERSION.dist-info/LICENSE          │ $VERSION.dist-info/licenses/LICENSE │
│ $VERSION.dist-info/top_level.txt    │ lumen/__version.py                  │
│ lumen/.version                      │ lumen/_version.py                   │
└─────────────────────────────────────┴─────────────────────────────────────┘

```
``` yaml
                                      Lumen - sdist
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Files only in run 1                          ┃ Files only in run 2                     ┃
┃ lumen 0.7.0a11                               ┃ lumen 0.7.0a11.post1.dev46+gce9c313     ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ $VERSION/MANIFEST.in                         │ $VERSION/.gitignore                     │
│ $VERSION/lumen.egg-info/PKG-INFO             │ $VERSION/lumen/__version.py             │
│ $VERSION/lumen.egg-info/SOURCES.txt          │ $VERSION/lumen/_version.py              │
│ $VERSION/lumen.egg-info/dependency_links.txt │ $VERSION/scripts/conda/build.sh         │
│ $VERSION/lumen.egg-info/entry_points.txt     │ $VERSION/scripts/conda/recipe/meta.yaml │
│ $VERSION/lumen.egg-info/requires.txt         │ -                                       │
│ $VERSION/lumen.egg-info/top_level.txt        │ -                                       │
│ $VERSION/lumen/.version                      │ -                                       │
│ $VERSION/setup.cfg                           │ -                                       │
│ $VERSION/setup.py                            │ -                                       │
└──────────────────────────────────────────────┴─────────────────────────────────────────┘

```
``` yaml
                                                 Lumen - conda #1
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Files only in run 1                                         ┃ Files only in run 2                               ┃
┃ lumen 0.7.0a11                                              ┃ lumen 0.7.0a11.post1.dev46+gce9c313               ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ site-packages/$VERSION-py3.12.egg-info/PKG-INFO             │ info/has_prefix                                   │
│ site-packages/$VERSION-py3.12.egg-info/SOURCES.txt          │ info/test/run_test.bat                            │
│ site-packages/$VERSION-py3.12.egg-info/dependency_links.txt │ info/test/run_test.sh                             │
│ site-packages/$VERSION-py3.12.egg-info/entry_points.txt     │ info/test/test_time_dependencies.json             │
│ site-packages/$VERSION-py3.12.egg-info/requires.txt         │ python-scripts/lumen                              │
│ site-packages/$VERSION-py3.12.egg-info/top_level.txt        │ python-scripts/lumen-ai                           │
│ site-packages/lumen/.version                                │ site-packages/$VERSION.dist-info/INSTALLER        │
│ -                                                           │ site-packages/$VERSION.dist-info/METADATA         │
│ -                                                           │ site-packages/$VERSION.dist-info/RECORD           │
│ -                                                           │ site-packages/$VERSION.dist-info/REQUESTED        │
│ -                                                           │ site-packages/$VERSION.dist-info/WHEEL            │
│ -                                                           │ site-packages/$VERSION.dist-info/direct_url.json  │
│ -                                                           │ site-packages/$VERSION.dist-info/entry_points.txt │
│ -                                                           │ site-packages/$VERSION.dist-info/licenses/LICENSE │
│ -                                                           │ site-packages/lumen/__version.py                  │
│ -                                                           │ site-packages/lumen/_version.py                   │
└─────────────────────────────────────────────────────────────┴───────────────────────────────────────────────────┘

```

### CODE REVIEW

File: lumen/tests/sources/test_duckdb.py [hunk 1]
```diff
@@ -8,10 +8,11 @@
 
 try:
     from lumen.sources.duckdb import DuckDBSource
+    pytestmark = pytest.mark.xdist_group("duckdb")
 except ImportError:
     DuckDBSource = None
```
Comment (hoxbro) [inline lumen/tests/sources/test_duckdb.py:13]:
```suggestion
```

File: pixi.toml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+[project]
+name = "lumen"
+channels = ["pyviz/label/dev", "bokeh", "conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+
+[tasks]
+# download-data = 'python scripts/download_data.py'
```
Comment (hoxbro) [inline pixi.toml:7]:
```suggestion
```

File: pixi.toml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+# nbformat = "*"
+# openai = "*"
+# pyarrow = "*"
+# pydantic = ">=2.8.0"
+#
+# [feature.ai-local.dependencies]
+# huggingface_hub = "*"
+#
+# [feature.ai-llama.dependencies]
+# llama-cpp-python = "*"
```
Comment (hoxbro) [inline pixi.toml:86]: I don't really know what to do with these. As far as I can see, they were never used in the CI. 

They are still available as optional dependencies.

File: pixi.toml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+# =============================================
+# =================== DOCS ====================
+# =============================================
+[feature.doc.dependencies]
+nbsite = ">=0.8.4,<0.9.0"
+
+[feature.doc.tasks]
+_docs-refs = 'python doc/generate_reference.py'
+_docs-generate = 'nbsite build --what=html --output=builtdocs --org holoviz --project-name lumen'
+#
```
Comment (hoxbro) [inline pixi.toml:97]:
```suggestion

```

File: pyproject.toml [hunk 5]
```diff
@@ -1,12 +1,99 @@
...
+xfail_strict = true
+log_cli_level = "INFO"
+filterwarnings = [
+    "error",
+    "ignore:datetime.datetime.utcfromtimestamp():DeprecationWarning:dateutil.tz.tz", # https://github.com/dateutil/dateutil/pull/1285
 ]
 
 [tool.ruff]
 fix = true
+target-version = "py37" # TODO: Remove so it is aligned with the project requires-python
```
Comment (hoxbro) [inline pyproject.toml:96]: I will remove this in a follow-up PR. If I do it now it will give some linting errors.

File: scripts/conda/build.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+set -euxo pipefail
+
+PACKAGE="lumen"
+
+python -m build -w .
+
+VERSION=$(python -c "import $PACKAGE; print($PACKAGE._version.__version__)")
+export VERSION
+
+# conda config --env --set conda_build.pkg_format 2
```
Comment (hoxbro) [inline scripts/conda/build.sh:12]: We can enable this after I have checked the output of the build artifact and the latest dev release.

File: pyproject.toml [hunk 7]
```diff
@@ -1,12 +1,99 @@
...
-    "param >=1.7.0",
-    "pyct >=0.4.4",
-    "setuptools >=30.3.0"
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "lumen"
+dynamic = ["version"]
+description = "A monitoring solution built on Panel."
```
Comment (hoxbro) [inline pyproject.toml:8]: @philippjfr Is this the description you want? (Copied from setup.py)

File: pyproject.toml [hunk 8]
```diff
@@ -1,12 +1,96 @@
...
+ai = ['nbformat', 'duckdb', 'pyarrow', 'openai', 'instructor >=1', 'pydantic >=2.8.0', 'datashader']
+ai-local = ['lumen[ai]', 'huggingface_hub']
+ai-llama = ['lumen[ai-local]', 'llama-cpp-python']
+
+[project.scripts]
+lumen = "lumen.command:main"
+lumen-ai = "lumen.command.ai:main"
+
+[project.entry-points.'panel.io.rest']
+lumen = "lumen.rest:lumen_rest_provider"
```
Comment (hoxbro) [inline pyproject.toml:59]: I need to double-check if this is also available in the conda package.

Comment (hoxbro) [reply pyproject.toml:59]: It works with conda, too.

File: pyproject.toml (hunk 5 above)
Comment (hoxbro) [reply pyproject.toml:93]: Done here: https://github.com/holoviz/lumen/pull/639

**Review by philippjfr — APPROVED** [review]

**Pull request merged**